### ISSUE TITLE
Template Specific *.11tydata.js Files Aren't Parsed

### ISSUE THREAD

**Issue by pengfluf** [11ty/eleventy#3072]
### Operating system

Linux

### Eleventy

2.0.1

### Describe the bug

Template (and directory) specific data `*.11ty.js` files aren't parsed/recognized.

### Reproduction steps

1. Having the following file structure, trying to attach the data to one of the includes
![1](https://github.com/11ty/eleventy/assets/25828285/74e571a1-5732-4411-9664-f81ea3e41fc2)
![2](https://github.com/11ty/eleventy/assets/25828285/14412d04-1ab3-4c04-9339-96ac7d37a369)
![3](https://github.com/11ty/eleventy/assets/25828285/b833b257-a1e6-46d3-ad1e-1506e7fd7fb2)
2. That include file is used in a layout
![4](https://github.com/11ty/eleventy/assets/25828285/53f38bca-48f2-4534-9d3a-fca3215ead96)
![5](https://github.com/11ty/eleventy/assets/25828285/7262d31b-5894-4661-917d-46dd504a8ef0)
3. The layout is used in src/index.njk entry file
![6](https://github.com/11ty/eleventy/assets/25828285/8c6b471a-2d72-4d26-a9d4-f9ff49bcc2bc)
4. Seeing only the first paragraph, the data is missing
![7](https://github.com/11ty/eleventy/assets/25828285/98f0d7ac-dd3d-4ea3-8097-5be9c6e63fad)

Tried other object key combinations to access the data - no luck.

Any ideas? 🤔

### Expected behavior

The data is available, the second paragraph is shown

**Comment by pdehaan** [comment]
Hhmm, it might be https://www.11ty.dev/docs/languages/nunjucks/#supported-features where front matter is not processed in front matter. Although maybe not 100% your use case and this is more specific to including a Nunjucks file isn't parse front matter from that included file, which, i can almost kinda sorta see making sense:

> FEATURE | SYNTAX
> -- | --
> ✅ Includes | `{% include 'included.njk' %}` looks in `_includes/included.njk`. Filenames must be in quotes. **Does not process front matter in the include file.**
> ✅ Includes (Relative Path) | Relative paths use `./ (template’s directory)` or `../ (template’s parent directory)`. Example: `{% include './included.njk' %}` looks for included.njk in the template’s current directory. **Does not process front matter in the include file.**

**Comment by pdehaan** [comment]
Not really a workaround, but one possible solution is to use the template data file in the src/index.njk file:

```sh
tree -A --gitignore
.
├── eleventy.config.js
├── package-lock.json
├── package.json
├── src/
│   ├── _includes/
│   │   ├── any-page.njk
│   │   └── test/
│   │       └── index.njk
│   ├── index.11tydata.js
│   └── index.njk
└── www/
    └── index.html

5 directories, 8 files
```

### OUTPUT
```html
<section>
  <p>visible text</p>
  <p>data</p>
</section>

<main>THis is some fancy content!</main>
```

https://github.com/pdehaan/11ty-3072

**Comment by pengfluf** [comment]
@pdehaan Thank you very much for a detailed response 🙂

As far as I understand the `include` files (`*.njk` ones at least) are meant to be without data - I wonder is there a way to include multiple template chunks into pages or layouts using local data much like it's possible in JS component frameworks these days? Or is there a different frame of thinking applied here?

**Comment by pdehaan** [comment]
@pengfluf I think if you add all your local data file to the src/index.njk (either in front matter, or directory/template/global data files) then the data should all be available from the `{% include %}`ed files.

**Comment by pengfluf** [comment]
Yeah defined the data in the same folder and then required it in the global data 😄

Would be nice though to have the include files/components isolated with their data and 11ty being able to recognize that data 🙂

Thank you for helping!
